### ISSUE TITLE
Buffer overflow from input anywhere in machine, hangs machine.

### ISSUE THREAD

**Issue by justinmarma** [Bone-Project/BoneOS#165]
Below is a linked image of the POC, system is hanged

<img src="http://image.prntscr.com/image/c2857c81f83d4ee888e46e8ed8112c7c.png"/>

**Comment by amanuel2** [comment]
That is because the [`get_opt` function](https://github.com/Bone-Project/BoneOS/blob/master/libc/unistd/get_opt/get_opt.c#L43) uses [`str_t`](https://github.com/Bone-Project/BoneOS/blob/master/include/libc/string/str_t.h#L8) as the type of the sting provided which has a maximum buffer of 200. @Fortunate-MAN will fix this as soon as i add memory manager and get it to char* .  

Note for @Fortunate-MAN  : 

I know in some places i do `for(int i=0; i<200; i++) str_t_instance[i]=0;` to clear out the buffer , dont forget to change those to fix this problem after i get memory manager working.

**Comment by double-fault** [comment]
@justinmarma Can you give the exact ways how to reproduce this problem on another computer? I would need information like which OS you are running, are you running the latest build or v0.0.1, what commands are you doing before you run the OS or does it happen if you just start the OS?

Also, as per the image you have given, it looks like you are running an old version, so please get the Nightly Build from bintray or compile from source and run it.

**Comment by amanuel2** [comment]
@Fortunate-MAN v0.0.1 or v0.0.2 .. i released the second version yesterday.

**Comment by justinmarma** [comment]
@Fortunate-MAN The bug exists in v0.0.2 too. I'm running the OS in VirtualBox and I showed above that it happens when input exceeds buffer len.. 

The buffer length is 200 and the BOF happens around 56 extra bytes relative to the end of the buffer.

This can be reproduced by creating a script or program to produce 200 chars and add the variable length to it, then pasting it into the console if you use VBox.

**Comment by double-fault** [comment]
@justinmarma no repro on this one either. I'm running QEmu on Ubuntu 14.04 and nothing much happened even after I entered _way_ more than 200 chars. @amanuel2 can you repro this issue?

![selection_143](https://cloud.githubusercontent.com/assets/12422882/22063765/a6a48496-dda6-11e6-8c31-e6dfe59dbbc1.png)

**Comment by amanuel2** [comment]
same here i cant reproduce this but, when you press enter the big crash happenes.

**Comment by justinmarma** [comment]
@amanuel2  @Fortunate-MAN The larger the input the more memory gets corrupted, the "big crash" is the buffer overflow and subsequent crash, so you are in fact reproducing the error, you're just not getting the same error I got because of the insane amount of chars you entered.

**Comment by justinmarma** [comment]
[Fresh Crash screenshot](http://prntscr.com/dxfv3y)    :) :) :) :)

**Comment by double-fault** [comment]
@justinmarma so you're saying that the crash happens _after_ you press enter like @amanuel2 said?

**Comment by double-fault** [comment]
@justinmarma please reply.

**Comment by justinmarma** [comment]
@Fortunate-MAN  Yes, the BOF can only happen if a large amount of data is written into a buffer smaller than the input, which doesn't happen until one presses "enter" and it's written to the buffer. I think the BOF is in str_t but I may be wrong.
